src/components/app/loading-component.tsx

```diff
@@ -3,13 +3,19 @@ import { Loader } from "react-feather";
 import { cn } from "@/lib/utils";
 
 export function ComponentIsLoading({ fullPage }: { fullPage: boolean }) {
+  const [delayed, setDelayed] = React.useState(true);
+
+  React.useEffect(() => {
+    setTimeout(() => setDelayed(false), 500);
+  }, []);
+
   return (
     <div
       className={cn("flex justify-center items-center w-full h-full", {
         "w-lvw h-lvh": fullPage,
       })}
     >
-      <Loader className="animate-spin w-8 h-8 text-primary" />
+      {!delayed && <Loader className="animate-spin w-8 h-8 text-primary" />}
```

Comment:
And it would also be nice if we could turn on and off the delayed-ness, not sure it is very loading component that will need it, so we should have room for those that wont need it:
a suggestion is that we have a <Delayed wait={500} />  the `wait` prop will have default value, This way we can have something like this
```js
<Delayed wait={500}>
<ComponentIsLoading />
</Delayed>
```

Reply:
Got it